### PR TITLE
Update PHP Sync Issue generation script to ignore PRs with given labels

### PR DESCRIPTION
**Pull request by getdave**
## What?

Updates the script which generates the list of PRs requiring manual backporting to WP Core for a given release to ignore PRs with known labels.

## Why?

As part of changes to the process for syncing Gutenberg to WP Core we now add a `Backported to WP Core` label to any PRs that have been backported. This is done either manually or [automatically via the Cherry Picking script](https://github.com/WordPress/gutenberg/pull/58970).

As a result, we should update the script to ignore any PRs that have this label as we _know_ they have already been backported to Core.

As a consequence we can also remove the `lastRcDate` required arg as this was a temp hack in place during 6.5 because the PRs from the 6.4 cycle didn't have the `Backported to WP Core` label  (as this convention was adopted only during the 6.5 release).

## How?

- Fetches full PR info for all commits 
- Checks labels and omits any PRs which don't need to be processed based on those labels
- Updates labels to include the `Backported to WP Core` label.
- A few other tweaks

## Testing Instructions

- Get GH token
- Run script:
```
node bin/generate-php-sync-issue.mjs --token=YOUR_TOKEN_HERE --since=2023-09-23  --wpstable=6.4
```
- Check that PRs listed do not have the `Backported to WP Core` label 
- Check that PRs listed do not have the `Backport from WordPress Core` label

### CODE REVIEW

File: bin/generate-php-sync-issue.mjs [hunk 1]
```diff
@@ -17,18 +17,22 @@ function getArg( argName ) {
 
 const OWNER = 'wordpress';
 const REPO = 'gutenberg';
-const MAX_MONTHS_TO_QUERY = 4;
+const MAX_MONTHS_TO_QUERY = 6;
 
 // The following paths will be ignored when generating the issue content.
 const IGNORED_PATHS = [
+	'init.php', // plugin specific code.
```
Comment (getdave) [inline bin/generate-php-sync-issue.mjs:24]: Added this to path ignore as I don't think we need to worry about code here.

File: bin/generate-php-sync-issue.mjs [hunk 2]
```diff
@@ -17,18 +17,22 @@ function getArg( argName ) {
...
 	'packages/e2e-tests/plugins', // PHP files related to e2e tests only.
-	'packages/block-library', // this is handled automatically.
+	'packages/block-library', // packages are synced to WP Core via npm packages.
 ];
 
 // PRs containing the following labels will be ignored when generating the issue content.
-const LABELS_TO_IGNORE = [ 'Backport from WordPress Core' ];
+const LABELS_TO_IGNORE = [
+	'Backport from WordPress Core', // PRs made "upstream" in Core that were synced back into Gutenberg.
+	'Backported to WP Core', // PRs that were synced into Core during a previous release.
```
Comment (getdave) [inline bin/generate-php-sync-issue.mjs:34]: This is the new label that indicates a given PR has already been backported to WP Core and so any PRs with this label can be safely omitted from the list of PRs that will need to be manually backported to Core in the _next_ release cycle (at the time of authoring that will be WP 6.6).

File: bin/generate-php-sync-issue.mjs [hunk 3]
```diff
@@ -466,7 +439,7 @@ function generateIssueContent( result, level = 1 ) {
 }
 
 async function fetchAllCommits( since, path ) {
-	return await octokitPaginate( 'GET /repos/{owner}/{repo}/commits', {
```
Comment (getdave) [inline bin/generate-php-sync-issue.mjs:469]: No need to await inside an async function as return value will always be an awaitable `Promise`.

File: bin/generate-php-sync-issue.mjs [hunk 4]
```diff
@@ -481,7 +454,7 @@ async function fetchCommit( sha ) {
 
 // eslint-disable-next-line no-unused-vars
 async function getPullRequestDataForCommit( commitSha ) {
-	const pullRequests = octokitRequest(
+	const pullRequests = await octokitRequest(
```
Comment (getdave) [inline bin/generate-php-sync-issue.mjs:446]: We need to await here to ensure we get the data 😅

**Comment by getdave** [comment]
@youknowriad Could we get this merged do you think? Without these changes this script will be less useful to future contributors and I'd like to update [the handbook](https://make.wordpress.org/core/handbook/about/release-cycle/block-editor-release-process-for-major-releases/).

**Review by youknowriad — APPROVED** [review]

**Pull request merged**